### PR TITLE
feat(882): Optimize metrics function usage during onboarding flow

### PR DESCRIPTION
**Pull request by DDDDDanica**
## Explanation
Only start `trackEvent` method on `ui/pages/onboarding-flow/welcome/welcome.js` when user opts in metrics track on onboarding.

### Before
**_Send metric events_**

https://github.com/MetaMask/metamask-extension/assets/12678455/9df12029-1b2a-44e3-9163-b5eb0885073a

### After
**_Will not send metric events if not opting in_**

https://github.com/MetaMask/metamask-extension/assets/12678455/877176bd-a084-41ae-ac62-c74c851d05aa

## Manual Testing Steps
- Open network tab when onboarding
- Check if there's any `batch` request to send metrics

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: test/e2e/helpers.js [hunk 1]
```diff
@@ -313,10 +313,10 @@ const onboardingBeginCreateNewWallet = async (driver) => {
  * Choose either "I Agree" or "No Thanks" on the MetaMetrics onboarding screen
  *
  * @param {WebDriver} driver
- * @param {boolean} optin - true to opt into metrics, default is false
+ * @param {boolean} option - true to opt into metrics, default is false
  */
-const onboardingChooseMetametricsOption = async (driver, optin = false) => {
```
Comment (DDDDDanica) [inline test/e2e/helpers.js:318]: Typo fix

File: test/e2e/helpers.js [hunk 2]
```diff
@@ -744,15 +744,17 @@ async function switchToNotificationWindow(driver) {
  *
  * @param {WebDriver} driver
  * @param {import('mockttp').Mockttp} mockedEndpoints
+ * @param {boolean} hasRequest
  * @returns {import('mockttp/dist/pluggable-admin').MockttpClientResponse[]}
  */
-async function getEventPayloads(driver, mockedEndpoints) {
+async function getEventPayloads(driver, mockedEndpoints, hasRequest = true) {
```
Comment (DDDDDanica) [inline test/e2e/helpers.js:750]: Expand method to test no request scenario as well

File: test/e2e/metrics/permissions-approved.spec.js [hunk 3]
```diff
@@ -43,28 +44,6 @@ async function mockSegment(mockServer) {
   ];
 }
 
-/**
```
Comment (DDDDDanica) [inline test/e2e/metrics/permissions-approved.spec.js:46]: Duplicated in helper.js

File: test/e2e/metrics/unlock-wallet.spec.js [hunk 4]
```diff
@@ -1,8 +1,12 @@
...
-const { convertToHexValue, withFixtures } = require('../helpers');
+const {
+  withFixtures,
+  unlockWallet,
+  defaultGanacheOptions,
+} = require('../helpers');
 const FixtureBuilder = require('../fixture-builder');
 
-describe('Segment metrics', function () {
+describe('Unlock wallet', function () {
```
Comment (DDDDDanica) [inline test/e2e/metrics/unlock-wallet.spec.js:9]: Move to `metrics` folder

File: test/e2e/metrics/unlock-wallet.spec.js [hunk 5]
```diff
@@ -32,34 +28,31 @@ describe('Segment metrics', function () {
...
-        assert.equal(path, '/');
+        assertBatchValue(firstMock, 'Home', '/');
+        assertBatchValue(secondMock, 'Unlock Page', '/unlock');
+        assertBatchValue(thirdMock, 'Home', '/');
       },
     );
   });
 });
+
+function assertBatchValue(mockRequest, assertedTitle, assertedPath) {
```
Comment (DDDDDanica) [inline test/e2e/metrics/unlock-wallet.spec.js:53]: Refactor to make it more readable

**Review by danjm — APPROVED** [review]
Nice work!

**Review by pedronfigueiredo — APPROVED** [review]

**Comment by pedronfigueiredo** [comment]
LGTM!

**Pull request merged**